src/direct-auth-idx/direct-auth-idx/Controllers/ManageController.cs

```diff
@@ -143,22 +114,20 @@ private async Task<ActionResult> VerifyAuthenticatorAsync(string code, string vi
             {
                 var authnResponse = await idxAuthClient.VerifyAuthenticatorAsync(verifyAuthenticatorOptions, (IIdxContext)Session["idxContext"]);
 
-                if (authnResponse.AuthenticationStatus == AuthenticationStatus.AwaitingPasswordReset)
+                switch (authnResponse.AuthenticationStatus)
                 {
-                    // TODO: Force authentication and redirect to home page
-                    Session["idxContext"] = authnResponse.IdxContext;
+                    case AuthenticationStatus.AwaitingPasswordReset:
+                        // TODO: Force authentication and redirect to home page
+                        Session["idxContext"] = authnResponse.IdxContext;
+                        return RedirectToAction("ChangePassword", "Manage");
 
-                    return RedirectToAction("ChangePassword", "Manage");
-                }
-                else if (authnResponse.AuthenticationStatus == AuthenticationStatus.AwaitingAuthenticatorEnrollment)
-                {
-                    Session["idxContext"] = authnResponse.IdxContext;
-                    TempData["authenticators"] = authnResponse.Authenticators;
-                    return RedirectToAction("selectAuthenticator", "Manage");
-                }
-                else if (authnResponse.AuthenticationStatus == AuthenticationStatus.Success)
-                {
-                    return RedirectToAction("Login", "Account");
+                    case AuthenticationStatus.AwaitingAuthenticatorEnrollment:
+                        Session["idxContext"] = authnResponse.IdxContext;
+                        TempData["authenticators"] = authnResponse.Authenticators;
+                        return RedirectToAction("selectAuthenticator", "Manage");
+
+                    case AuthenticationStatus.Success:
```

Comment:
If authenticationStatus is successful, let's use the helper to authenticate the user and redirect them to the home page instead.